src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -113,19 +117,35 @@ public static void setDatesToRepoConfigs(
      */
     public static void merge(List<RepoConfiguration> repoConfigs, List<RepoConfiguration> authorConfigs) {
         for (RepoConfiguration authorConfig : authorConfigs) {
-            int index = repoConfigs.indexOf(authorConfig);
+            List<RepoConfiguration> repoConfigsToAdd = new ArrayList<>();
 
-            if (index == -1) {
-                logger.warning(String.format(
-                        "Repository %s is not found in repo-config.csv.", authorConfig.getLocation()));
-                continue;
-            }
+            if (authorConfig.location.isEmpty()) {
+                repoConfigsToAdd = repoConfigs;
+            } else {
+                int index = repoConfigs.indexOf(authorConfig);
 
-            RepoConfiguration repoConfig = repoConfigs.get(index);
+                if (index == -1) {
+                    logger.warning(String.format(
+                            "Repository %s is not found in repo-config.csv.", authorConfig.getLocation()));
+                    continue;
+                }
 
-            repoConfig.setAuthorList(authorConfig.getAuthorList());
-            repoConfig.setAuthorDisplayNameMap(authorConfig.getAuthorDisplayNameMap());
-            repoConfig.setAuthorAliasMap(authorConfig.getAuthorAliasMap());
+                repoConfigsToAdd.add(repoConfigs.get(index));
+            }
+
+            for (RepoConfiguration repoConfigToAdd : repoConfigsToAdd) {
+                for (Author author: authorConfig.getAuthorList()) {
+                    if (repoConfigToAdd.containsAuthor(author)) {
+                        logger.warning(String.format(
+                                "Skipping author as %s already in repository %s",
+                                author.getGitId(), repoConfigToAdd.getDisplayName()));
+                        continue;
+                    }
+                    repoConfigToAdd.addAuthor(author);
+                    repoConfigToAdd.addAuthorAliases(author, author.getAuthorAliases());
+                    repoConfigToAdd.setAuthorDisplayName(author, authorConfig.getAuthorDisplayNameMap().get(author));
+                }
+            }
```

Comment:
I believe this method can be further simplified. 

`i.e.` 
1. Keep current code but track a list of `authorConfigs` with no locations
1. After the for loop in current code,  add those `authorConfigs`. This should reduce the number of loops.
1. Update methods like `addAuthor`, `addAuthorAliases`, ... to prevent duplicates.
1. Apply SLAP - `https://se-edu.github.io/se-book/codeQuality/maximiseReadability/intermediate/slapHard/index.html`.

Reply:
OK, I will try to find a way to simplify it